evmcore/tx_validation_test.go

```diff
@@ -16,68 +16,71 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+// getTestTransactionsOption returns a set of options to adjust the validation of transactions
+// so that it would accept all types of transactions, considering them as local transactions
+// with a min tip of 1, current base fee of 1, and a current max gas of 100_000.
+func getTestTransactionsOption() (validationOptions, networkOptions) {
+	return validationOptions{
+			minTip:  big.NewInt(1),
+			isLocal: true,
+			signer:  types.NewPragueSigner(big.NewInt(1)),
+		}, networkOptions{
+			eip2718:        true,
+			eip1559:        true,
+			shanghai:       true,
+			eip4844:        true,
+			eip7623:        true,
+			eip7702:        true,
+			currentBaseFee: big.NewInt(1),
+			currentMaxGas:  100_000,
+		}
+}
+
 func TestValidateTx_BeforeEip2718_RejectsNonLegacyTransactions(t *testing.T) {
 	for name, tx := range getTxsOfAllTypes() {
 		if _, ok := tx.(*types.LegacyTx); ok {
 			continue // Skip legacy transactions
 		}
 		t.Run(name, func(t *testing.T) {
-			err := validateTx(types.NewTx(tx), validationOptions{eip2718: false})
-			require.ErrorIs(t, ErrTxTypeNotSupported, err)
+			opt, netOpts := getTestTransactionsOption()
+			netOpts.eip2718 = false
+			err := validateTx(types.NewTx(tx), opt, netOpts)
+			require.ErrorContains(t, err, "network validation failed")
```

Comment:
This seems like now the test is accepting any reason for failing the validation while before the test was specifically targeting the "ErrTxTypeNotSupported" issue.